File: 374.Guess Number Higher or Lower.cpp
/**
 *Result:
 * 25 / 25 test cases passed.
 * Status: Accepted
 * Runtime: 0 ms
 * Your runtime beats 4.57% of cpp submissions.
 *Date:
 * 10/17/2016
 */

// Forward declaration of guess API.
// @param num, your guess
// @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        int l = 1;
        int r = n;
        while(l<r){
            int mid = l+(r-l)/2;    //avoid overflow
            int res = guess(mid);
            if(res==0)
                return mid;
            else if(res==1){
                l = mid+1;
            }else{
                r = mid-1;
            }
        }
        return l;
    }
};

File: 374.Guess Number Higher or Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        start, end = 1, n + 1
        while start != end:
            mid = start + (end - start) / 2
            res = guess(mid)
            if res == 0:
                return mid
            if res > 0:
                start = mid + 1
            else:
                end = mid
        return start
